### DISCUSSION TITLE
Why R code chunks are rendered, if i already rendered them?

### DISCUSSION THREAD

**Discussion by rogerio-bio** [quarto-dev/quarto-cli#4115] [Q&A] [answered]
I have already rendered the codes in R, and it takes me a whole day to do it. I have to choose the model that should go to the next step, so I did it step by step. Now, when it comes time to render the PDF, the Quarto is simply running everything again from scratch..... Is there any way to prevent this? I want just to render the PDF with the results and images? Do not rerun everything from scratch....

**Comment by baptiste** [comment, marked as answer]
It's the idea of reproducible documents that the code be run when the document is rendered. But for lengthy computations you can store results using the [cache mechanism](https://quarto.org/docs/projects/code-execution.html#cache). 

If you've run the code manually and stored the results, e.g as .rds files, and don't want to re-run it even once in Quarto (with cache enabled), you may have to load the .rds manually in your code, e.g 

```
if(file.exists("myresults.rds"))
  results = readRDS("myresults.rds")
```

**Reply by rogerio-bio** [reply]
Thank you 💯 ! I will try this 👍
